packages/react-debug-tools/src/__tests__/ReactHooksInspection-test.js

```diff
@@ -335,4 +335,50 @@ describe('ReactHooksInspection', () => {
       ]);
     });
   });
+
+  it('should gracefully handle custom hooks that have the same name as primitive hooks', () => {
```

Comment:
Doesn't trigger the original runtime error _but_ the following simpler test generates an invalid parsed stack:
```js
function useState(value) {
  React.useState(value);
}

function Test() {
  useState(0);
  return null;
}

const tree = ReactDebugTools.inspectHooks(Test, {});
expect(tree).toEqual([
  {
    id: null,
    isStateEditable: false,
    name: "State",
    subHooks: [
      {
        id: 0,
        isStateEditable: true,
        name: "State",
        subHooks: [],
        value: 0,
      },
    ],
    value: undefined,
  },
]);
```